server/src/main/scala/controllers/Application.scala

```diff
@@ -180,6 +181,56 @@ class Application @Inject()(implicit val config: Configuration, env: Environment
     SDate(date.millisSinceEpoch - oneDayInMillis)
   }
 
+  def featureGuides(): Action[AnyContent] = Action.async { _ =>
+    val featureGuidesJson: Future[String] = FeatureGuideRow.getAll()
+    featureGuidesJson.map(Ok(_))
+  }
+
+  def isNewFeatureAvailableSinceLastLogin = Action.async { implicit request =>
+    val userEmail = request.headers.get("X-Auth-Email").getOrElse("Unknown")
+    val latestFeatureDateF: Future[Option[Timestamp]] = FeatureGuideRow.selectAll.map(_.headOption.map(_.uploadTime))
+    val latestLoginDateF: Future[Option[Timestamp]] = ctrl.userService.selectUser(userEmail.trim).map(_.map(_.latest_login))
+    for {
+      latestFeatureDate <- latestFeatureDateF
+      latestLoginDate <- latestLoginDateF
+    } yield (latestFeatureDate, latestLoginDate) match {
+      case (Some(featureDate), Some(loginDate)) =>
+        Ok(featureDate.after(loginDate).toString)
+      case _ =>
+        Ok(false.toString)
+    }
+  }
+
+  def recordFeatureGuideView(filename: String): Action[AnyContent] = authByRole(BorderForceStaff) {
+    Action.async { implicit request =>
+      val userEmail = request.headers.get("X-Auth-Email").getOrElse("Unknown")
+      FeatureGuideRow.getGuideIdForFilename(filename).flatMap {
+        case Some(id) =>
+          FeatureGuideViewRow
+            .insertOrUpdate(id, userEmail)
+            .map(_ => Ok(s"File $filename viewed updated"))
+        case None =>
+          Future.successful(Ok(s"File $filename viewed not updated as file not found"))
+      }
+    }
+  }
+
+  def viewCountByUser(): Action[AnyContent] = authByRole(BorderForceStaff) {
+    Action.async { implicit request =>
+      import spray.json._
+      val userEmail = request.headers.get("X-Auth-Email").getOrElse("Unknown")
+      implicit val stringListFormat: JsonFormat[Seq[String]] = new JsonFormat[Seq[String]] {
```

Comment:
There's an existing built-in `Seq` json format:

`import spray.json.DefaultJsonProtocol.{StringJsonFormat, immSeqFormat}`